src/arch/x86_64/kernel/gdt.rs

```diff
@@ -47,8 +49,10 @@ pub fn add_current_core() {
 			BasePageSize::SIZE as usize
 		};
 
-		let ist = crate::mm::allocate(sz, true);
-		let ist_start = ist.as_u64() + sz as u64 - TaskStacks::MARKER_SIZE as u64;
+		let layout = Layout::from_size_align(sz, BasePageSize::SIZE as usize).unwrap();
+		let ist = PhysAlloc.allocate(layout).unwrap();
+		let ist: u64 = ist.as_ptr().expose_provenance().try_into().unwrap();
+		let ist_start = ist + sz as u64 - TaskStacks::MARKER_SIZE as u64;
```

Comment:
Would offset methods on pointers be suitable here? Not a substantial change but I think it would reflect the meaning of what is happening here better. Otherwise, I think we can get the total first and then cast to u64 rather than casting each summand to `u64` and adding, since all are usize.